eng/common/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,256 @@
+parameters:
+  - name: SpecRepoUrl
+    type: string
+  - name: SdkRepoUrl
+    type: string
+  - name: SpecRepoCommit
+    type: string
+    default: '$(Build.SourceBranch)'
+  - name: SdkRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'API specification type'
+  - name: ConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to API specification file (TypeSpec/OpenAPI)'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages:
+    - stage: Build
+      displayName: 'SDK Generation'
```

Comment:
This is first part change, referencing the common template and defining the stages.